docs/language/overview.md

```diff
@@ -482,16 +480,15 @@ a few examples:
 * a union of string and integer  - `(string,int64)`
 
 Complex types may be composed, as in `[({s:string},{x:int64})]` which is
-an array of type union of two types of records.
+an array of type `union` of two types of records.
 
-The [typeof function](functions/typeof.md) returns a value's type as
+The [`typeof` function](functions/typeof.md) returns a value's type as
 a value, e.g., `typeof(1)` is `<int64>` and `typeof(<int64>)` is `<type>`.
 
 First-class types are quite powerful because types can
-serve as group-by keys or be used in "data shaping" logic.
-A common query for data introspection is to perform some search query
-slicing and dicing some exploratory data then counting the shapes of
-each type of data as follows:
+serve as group-by keys or be used in ["data shaping"](#9-shaping) logic.
+A common workflow for data introspection is to first perform a search of
+exploratory data and then count the shapes of each type of data as follows:
```

Comment:
The original statement felt a bit too wordy, so I've tried to simplify. Please confirm I have not wrecked the essence of its intended meaning, though.